catalyst/callbacks/sklearn_classifier.py

```diff
@@ -0,0 +1,189 @@
+                verbose=False,
+                logdir="./logs",
+                valid_loader="valid",
+                valid_metric="sklearn_classifier_metric",
+                minimize_valid_metric=False,
+                num_epochs=100,
+            )
+    """
+
+    def __init__(
+        self,
+        feature_key: str,
+        target_key: str,
+        train_loader: str,
+        valid_loader: str,
+        sklearn_classifier_fn: Callable,
+        predict_method: str = "predict",
+        predict_key: str = "sklearn_predict",
+    ) -> None:
+        super().__init__(order=CallbackOrder.Metric)
+        assert hasattr(
+            sklearn_classifier_fn(), predict_method
+        ), "The classifier must have the predict method!"
+        self._train_loader = train_loader
+        self._valid_loader = valid_loader
+        self.classifier_fabric = sklearn_classifier_fn
+        self.feature_key = feature_key
+        self.target_key = target_key
+        self.predict_method = predict_method
+        self.predict_key = predict_key
+        self.storage = AccumulationMetric(accumulative_fields=[feature_key, target_key])
+        self.classifier = None
+
+    def on_loader_start(self, runner: "IRunner") -> None:
+        """
+        Loader start hook: initiliaze storages for the loaders.
+
+        Args:
+            runner: current runner
+        """
+        super().on_loader_start(runner)
+        if runner.loader_key == self._train_loader:
+            self.storage.reset(
+                num_samples=runner.loader_batch_size * runner.loader_batch_len,
+                num_batches=runner.loader_batch_len,
+            )
+        if runner.loader_key == self._valid_loader:
+            assert self.classifier is not None, "The train loader has to be processed first!"
+
+    def on_batch_end(self, runner: "IRunner") -> None:
+        """On batch end action: get data from runner's batch and update a loader storage with it
+
+        Args:
+            runner: runner for the experiment.
+        """
+        if runner.loader_key == self._train_loader:
+            self.storage.update(**runner.batch)
+        if runner.loader_key == self._valid_loader:
+            features = runner.batch[self.feature_key].detach().cpu().numpy()
+            # classifier predict
+            classifier_predict = getattr(self.classifier, self.predict_method)
+            runner.batch[self.predict_key] = torch.Tensor(classifier_predict(features)).to(
+                runner.engine.device
+            )
+
+    def on_loader_end(self, runner: "IRunner") -> None:
+        """
+        Loader end hook: for the train loader train classifier/for the test check the quality
+
+        Args:
+            runner: current runner
+        """
+        if runner.loader_key == self._train_loader:
+            data = self.storage.compute_key_value()
+            # classifier fit
+            features, targets = data[self.feature_key].numpy(), data[self.target_key].numpy()

```

Comment:
```suggestion
            features, targets = data[self.feature_key].detach().cpu().numpy(), data[self.target_key].detach().cpu().numpy()

```

Reply:
AccumulationMetric does in the [update method](https://github.com/catalyst-team/catalyst/blob/master/catalyst/metrics/_metric.py#L235-L240).
But maybe just for simplicity, we can do it.